src/components/Clock.tsx

```diff
@@ -77,45 +75,33 @@ export function Clock({
 }: ClockProps) {
   const font = useFont(require('../fonts/digital-7.ttf'), 30);
 
-  const [currentSeconds, setCurrentSeconds] = useState(new Date().getSeconds());
-  const [currentMinutes, setCurrentMinutes] = useState(new Date().getMinutes());
-  const [currentHours, setCurrentHours] = useState(new Date().getHours());
-
-  const secondsRotation = useComputedValue(() => {
-    return [{ rotate: degreesToRadians(currentSeconds * 6) }];
-  }, [currentSeconds]);
-
-  const minutesRotation = useComputedValue(() => {
-    return [{ rotate: degreesToRadians(currentMinutes * 6) }];
-  }, [currentMinutes]);
-
-  const hoursRotation = useComputedValue(() => {
-    return [{ rotate: degreesToRadians(currentHours * 30) }];
-  }, [currentHours]);
+  const [date, setDate] = useState(new Date());
 
   useEffect(() => {
-    const increaseSeconds = setInterval(() => {
-      setCurrentSeconds((val) => val + 1);
+    const mainInterval = setInterval(() => {
+      setDate(new Date());
     }, ONE_SECOND_IN_MS);
+    return () => {
+      clearInterval(mainInterval);
+    };
+  }, []);
 
-    const increaseMinutes = setInterval(() => {
-      setCurrentMinutes((val) => val + 1);
-    }, ONE_MINUTE_IN_MS);
+  const secondsRotation = useComputedValue(() => {
+    return [{ rotate: degreesToRadians(date.getSeconds() * 6) }];
+  }, [date]);
 
-    const increaseHours = setInterval(() => {
-      setCurrentHours((val) => val + 1);
-    }, ONE_HOUR_IN_MS);
+  const minutesRotation = useComputedValue(() => {
+    return [{ rotate: degreesToRadians(date.getMinutes() * 6) }];
+  }, [date]);
 
-    return () => {
-      clearInterval(increaseSeconds);
-      clearInterval(increaseMinutes);
-      clearInterval(increaseHours);
-    };
-  }, [setCurrentSeconds]);
+  const hoursRotation = useComputedValue(() => {
+    return [{ rotate: degreesToRadians(date.getHours() * 30) }];
+  }, [date]);
 
   if (!font) return null;
 
   return (
+    // @ts-expect-error Canvas isn't a valid JSX element...
```

Comment:
`expect` throws me an error locally when checking the types, s hence changing it to `ignore`. This is due to the `react-native-skia` version, but will address it in another PR.

```suggestion
    // @ts-ignore Canvas isn't a valid JSX element...
```